### DISCUSSION TITLE
公式显示失败

### DISCUSSION THREAD

**Discussion by Bourne98** [tamlok/vnote#2127] [Q&A]
我之前写好了笔记，里面有很多公式，当时是可以成功显示的。但是今天打开软件以后，所有公式都无法正常显示，只显示源码。我换另一个软件读同样的md文件则没有问题。

**Comment by tamlok** [comment]
右上角有日志，查看一下？阅读模式不能显示，还是编辑模式不能显示？网络有没有问题？

**Reply by Bourne98** [reply]
都没有显示。而且我之前写好的公式现在也无法查看了。
下面是vnotex.log
`Info:(main.cpp:139) "VNote (v3.13.0) started at 周一 5月 16 20:18:26 2022 (windows)"
Info:(main.cpp:144) OpenSSL build version: "OpenSSL 1.1.1g  21 Apr 2020" link version: "OpenSSL 1.1.1j  16 Feb 2021"
Warning:(main.cpp:148) versions of the built and linked OpenSSL mismatch, network may not work
Info:(main.cpp:204) locale: "zh_CN"
Info:(thememgr.cpp:91) using theme "pure"
Warning:(:0) No language dictionaries for the language: "zh_CN" trying to load en_US as default
Info:(singleinstanceguard.cpp:139) local server receives new connect QLocalSocket(0x206eeb61840)
Info:(singleinstanceguard.cpp:139) local server receives new connect QLocalSocket(0x206eeb60fa0)
Info:(singleinstanceguard.cpp:24) guarding is off
Info:(main.cpp:139) "VNote (v3.13.0) started at 周一 5月 16 20:48:41 2022 (windows)"
Info:(main.cpp:144) OpenSSL build version: "OpenSSL 1.1.1g  21 Apr 2020" link version: "OpenSSL 1.1.1j  16 Feb 2021"
Warning:(main.cpp:148) versions of the built and linked OpenSSL mismatch, network may not work
Info:(main.cpp:204) locale: "zh_CN"
Info:(thememgr.cpp:91) using theme "pure"
Warning:(:0) No language dictionaries for the language: "zh_CN" trying to load en_US as default
Info:(singleinstanceguard.cpp:139) local server receives new connect QLocalSocket(0x22dc6ca45a0)
Critical:(markdownviewer.js:155) MathJax is not ready TypeError: MathJax.texReset is not a function
Critical:(markdownviewer.js:111) failed to render MathJax TypeError: MathJax.getMetricsFor is not a function
Warning:(markdownviewer.js:146) failed to preview graph 0 10
Critical:(markdownviewer.js:111) failed to render MathJax TypeError: MathJax.getMetricsFor is not a function
Warning:(markdownviewer.js:146) failed to preview graph 0 11
Critical:(markdownviewer.js:155) MathJax is not ready TypeError: MathJax.texReset is not a function
Info:(singleinstanceguard.cpp:24) guarding is off
Info:(main.cpp:139) "VNote (v3.13.0) started at 周一 5月 16 22:12:24 2022 (windows)"
Info:(main.cpp:144) OpenSSL build version: "OpenSSL 1.1.1g  21 Apr 2020" link version: "OpenSSL 1.1.1j  16 Feb 2021"
Warning:(main.cpp:148) versions of the built and linked OpenSSL mismatch, network may not work
Info:(main.cpp:204) locale: "zh_CN"
Info:(thememgr.cpp:91) using theme "pure"
Warning:(:0) No language dictionaries for the language: "zh_CN" trying to load en_US as default
Critical:(markdownviewer.js:111) failed to render MathJax TypeError: MathJax.getMetricsFor is not a function
Warning:(markdownviewer.js:146) failed to preview graph 0 1
`

**Reply by MOWAGN** [reply]
公式显示功能需要网络吗？无网状态是不是无法转换成公式？

**Reply by tamlok** [reply]
> 公式显示功能需要网络吗？无网状态是不是无法转换成公式？

可以本地化，看这里的上下文或者文档

**Comment by tamlok** [comment]
marhjax配置有更改吗？不行就右上角设置了里面重置一下吧

**Reply by Bourne98** [reply]
设置里的重置是灰色的，应该是我没有更改设置（我试过删除用户数据也没用）

**Reply by Bourne98** [reply]
额，同样的情况还出现在了我的ubuntu系统，我再找找其他的原因

**Reply by Bourne98** [reply]
您好，我查看了vnotex.json中的内容，当前有mathjax
关键字的部分内容如下

"mathjax_script": ""

"name": "mathjax",
"scripts": [
"web/js/mathjax.js"
],

这方面我不是很了解，您看着是否有什么问题

**Comment by Bourne98** [comment]
您好，我查看了vnotex.json中的内容，当前有mathjax
关键字的部分内容如下           

"mathjax_script": ""


"name": "mathjax",
   "scripts": [
                            "web/js/mathjax.js"
                        ],

这方面我不是很了解，您看着是否有什么问题

**Reply by yui525** [reply]
您好，我也出现了同样的问题，您解决了吗？

**Reply by tamlok** [reply]
> 您好，我也出现了同样的问题，您解决了吗？

可能是用的cdn挂了，搜一下issue，设置里面给一个新的xdn的值即可

**Reply by yui525** [reply]
oooook

**Reply by 6xmim** [reply]
应该是网络问题

**Reply by Bourne98** [reply]
> oooook
您解决了吗。我这还没有，您怎么修改的啊？请教您一下，我没看懂怎么修改xdn

**Reply by Bourne98** [reply]
> > 您好，我也出现了同样的问题，您解决了吗？
> 
> 可能是用的cdn挂了，搜一下issue，设置里面给一个新的xdn的值即可

您解决了吗。我这还没有，您怎么修改的啊？请教您一下，我没看懂怎么修改xdn，谢谢您

**Reply by yui525** [reply]
在这个页面上方的“Issues”里有些讨论mathjax的贴。这个问题大概就是软件默认的mathjax网络渲染的地址挂了。我是在贴里找了一个可用的脚本。具体方法是：在`设置`-`编辑器`-`Markdown编辑器`-`覆盖MathJax脚本`中输入`https://cdn.bootcdn.net/ajax/libs/mathjax/3.1.2/es5/tex-svg.js`。重启软件就好了。参考这个https://github.com/vnotex/vnote/issues/2133

**Reply by 6xmim** [reply]
> > > 您好，我也出现了同样的问题，您解决了吗？
> > 
> > 
> > 可能是用的cdn挂了，搜一下issue，设置里面给一个新的xdn的值即可
> 
> 您解决了吗。我这还没有，您怎么修改的啊？请教您一下，我没看懂怎么修改xdn，谢谢您

可以自定义本地的 mathjax 解决这个问题，参考官方文档
![image](https://user-images.githubusercontent.com/40284084/170898019-348f029b-97f5-4256-ab1b-ba7692303f55.png)

**Reply by Bourne98** [reply]
> > > > 您好，我也出现了同样的问题，您解决了吗？
> > > 
> > > 
> > > 可能是用的cdn挂了，搜一下issue，设置里面给一个新的xdn的值即可
> > 
> > 
> > 您解决了吗。我这还没有，您怎么修改的啊？请教您一下，我没看懂怎么修改xdn，谢谢您
> 
> 可以自定义本地的 mathjax 解决这个问题，参考官方文档 ![image](https://user-images.githubusercontent.com/40284084/170898019-348f029b-97f5-4256-ab1b-ba7692303f55.png)
解决了，谢谢您

**Reply by Bourne98** [reply]
> 在这个页面上方的“Issues”里有些讨论mathjax的贴。这个问题大概就是软件默认的mathjax网络渲染的地址挂了。我是在贴里找了一个可用的脚本。具体方法是：在`设置`-`编辑器`-`Markdown编辑器`-`覆盖MathJax脚本`中输入`https://cdn.bootcdn.net/ajax/libs/mathjax/3.1.2/es5/tex-svg.js`。重启软件就好了。参考这个#2133

好的，解决了，谢谢您

**Comment by tamlok** [comment]
这个没问题，你试试检查更新看网络行不行。
